src/util/prefix_filter.h

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************\
+
+Module: Prefix Filtering
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Prefix Filtering
+
+#ifndef CPROVER_UTIL_PREFIX_FILTER_H
+#define CPROVER_UTIL_PREFIX_FILTER_H
+
+#include <string>
+#include <vector>
+
+class prefix_filtert
+{
+public:
+  prefix_filtert(
+    std::vector<std::string> &&included_prefixes,
+    std::vector<std::string> &&excluded_prefixes);
+
+  /// Return true iff \p value matches a prefix in `included_prefixes`,
+  /// but doesn't match a prefix in `excluded_prefixes`.
+  /// An empty vector of `included_prefixes` is treated as 'match all'.
+  /// An empty vector of `excluded_prefixes` is treated as 'match nothing'.
+  bool operator()(const std::string &value);
```

Comment:
:bulb: To avoid the clunky calling syntax, you might just want to make this a free function that operators on a prefix_filter so you can give it a name.

Reply:
I agree with @thk123, the fields of this class are just inputs of the function and there is no invariant preserved by this class. I would prefer a struct with two public field, and a free function. That would be easier to use.